File: bindings/node/__tests__/env.test.js
const test = require('node:test')
const assert = require('node:assert')
const { Env } = require('..')

const SPEC = JSON.stringify({
  dataset_ref: 'smoke',
  symbol: 'TEST',
  observation: { features: [{ kind: 'price', field: 'close' }] },
  action_space: { type: 'discrete', n: 3 },
  reward: 'pnl',
  episode: { max_steps: 100, warmup: 0 },
})

const CANDLES = Array.from({ length: 5 }, (_, i) => ({
  ts: i,
  open: 100 + i,
  high: 100 + i,
  low: 100 + i,
  close: 100 + i,
}))

test('load, reset and step', () => {
  const env = new Env(SPEC)
  assert.strictEqual(env.command(JSON.stringify({ cmd: 'load', candles: CANDLES })), '{"ok":true}')
  const reset = JSON.parse(env.command(JSON.stringify({ cmd: 'reset' })))
  assert.deepStrictEqual(reset.observation, [100.0])
  const step = JSON.parse(env.command(JSON.stringify({ cmd: 'step', action: 2 })))
  assert.strictEqual(step.reward, 1.0)
  assert.strictEqual(step.terminated, false)
  assert.strictEqual(step.info.step, 1.0)
})

test('version is a string', () => {
  const env = new Env(SPEC)
  assert.strictEqual(typeof env.version(), 'string')
})

test('a bad spec throws', () => {
  assert.throws(() => new Env(JSON.stringify({ not: 'a spec' })))
})

File: bindings/java/src/test/java/org/wickra/gym/EnvTest.java
package org.wickra.gym;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

class EnvTest {
    private static final String SPEC =
            "{\"dataset_ref\":\"smoke\",\"symbol\":\"TEST\","
                    + "\"observation\":{\"features\":[{\"kind\":\"price\",\"field\":\"close\"}]},"
                    + "\"action_space\":{\"type\":\"discrete\",\"n\":3},"
                    + "\"reward\":\"pnl\",\"episode\":{\"max_steps\":100,\"warmup\":0}}";

    private static String loadCommand() {
        StringBuilder sb = new StringBuilder("{\"cmd\":\"load\",\"candles\":[");
        for (int i = 0; i < 5; i++) {
            double p = 100.0 + i;
            if (i > 0) {
                sb.append(',');
            }
            sb.append("{\"ts\":").append(i)
                    .append(",\"open\":").append(p)
                    .append(",\"high\":").append(p)
                    .append(",\"low\":").append(p)
                    .append(",\"close\":").append(p).append('}');
        }
        return sb.append("]}").toString();
    }

    @Test
    void versionIsNonEmpty() {
        assertFalse(Env.version().isEmpty());
    }

    @Test
    void loadResetStep() {
        try (Env env = new Env(SPEC)) {
            assertEquals("{\"ok\":true}", env.command(loadCommand()));
            String reset = env.command("{\"cmd\":\"reset\"}");
            assertTrue(reset.contains("\"observation\""), reset);
            String step = env.command("{\"cmd\":\"step\",\"action\":2}");
            assertTrue(step.contains("\"reward\":1.0"), step);
            assertTrue(step.contains("\"terminated\":false"), step);
        }
    }

    @Test
    void badSpecThrows() {
        assertThrows(IllegalArgumentException.class, () -> new Env("{\"not\":\"a spec\"}"));
    }
}
